testing/trino-tests/src/test/java/io/trino/tests/ci/TestCiWorkflow.java

```diff
@@ -96,6 +98,52 @@ public void testUploadTestResultsCondition()
                 .containsExactlyInAnyOrderElementsOf(testStepIds);
     }
 
+    @Test
+    public void testBuildSuccessDependencies()
+            throws Exception
+    {
+        String buildSuccessJobName = "build-success";
+
+        Yaml yaml = new Yaml();
+        Map<?, ?> workflow = yaml.load(new StringReader(Files.readString(findRepositoryRoot().resolve(CI_YML_REPO_PATH))));
+        Map<String, ?> jobs = getMap(workflow, "jobs");
+
+        Set<String> allJobNames = jobs.keySet();
+        assertThat(allJobNames).as("allJobNames").contains(buildSuccessJobName);
+        List<String> testJobNames = allJobNames.stream().filter(Predicate.not(buildSuccessJobName::equals))
+                .sorted()
+                .toList();
+        Map<?, ?> buildSuccessJob = getMap(jobs, buildSuccessJobName);
+
+        List<String> buildSuccessDependencies = getList(buildSuccessJob, "needs").stream()
+                .map(String.class::cast)
+                .collect(toImmutableList());
+        assertThat(buildSuccessDependencies).as("dependencies for %s", buildSuccessJobName)
+                .isSorted()
+                .doesNotHaveDuplicates()
+                .containsExactlyElementsOf(testJobNames);
+
+        StringBuilder expectedRunDefinition = new StringBuilder();
+        expectedRunDefinition.append("# generated by %s\n".formatted(getClass().getSimpleName()));
+        testJobNames.stream()
+                .map("echo '${{ needs.%1$s.result }}' | grep -xE 'success|skipped' || { echo 'Job \"%1$s\" failed' >&2; exit 1; }\n"::formatted)
+                .forEachOrdered(expectedRunDefinition::append);
+        Map<?, ?> runDefinition = getList(buildSuccessJob, "steps").stream()
+                .map(step -> (Map<?, ?>) step)
+                .filter(step -> "Check results".equals(ifPresent(step, "name", TestCiWorkflow::getString)))
```

Comment:
Since this is calling `String.equals()` which handles both null and foreign classes, no need for the method:
```suggestion
                .filter(step -> "Check results".equals(step.get("name")))
```